main.go

```diff
@@ -43,11 +45,31 @@ func newMetrics(reg prometheus.Registerer) *metrics {
 			Name: "govuk_mirror_last_updated_time",
 			Help: "Last time the mirror was updated",
 		}, []string{"backend"}),
+		mirrorResponseStatusCode: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Name: "govuk_mirror_response_status_code",
+			Help: "Response status code for the availability the mirror",
+		}, []string{"backend"}),
 	}
 	reg.MustRegister(m.mirrorLastUpdatedGauge)
+	reg.MustRegister(m.mirrorResponseStatusCode)
 	return m
 }
 
+func fetchMirrorAvailabilityMetric(backend string, url string) (int, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return -1, err
+	}
+	req.Header.Set("Backend-Override", backend)
+
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return -1, err
+	}
+
+	return resp.StatusCode, nil
+}
```

Comment:
optional: naming the return values here could help to make them a bit more self-documenting here, e.g. make it clear at a glance that the int is an HTTP status. Like this:
```suggestion
func fetchMirrorAvailabilityMetric(backend string, url string) (httpStatus int, err error) {
	req, err := http.NewRequest("GET", url, nil)
	if err != nil {
		return
	}
	req.Header.Set("Backend-Override", backend)

	resp, err := httpClient.Do(req)
	if err != nil {
		return
	}

	return resp.StatusCode, nil
}
```